### PR TITLE
Renaming container stream api and improving documentation example throughout code

### PR DESCRIPTION
**Pull request by simplynaveen20**
This PR will remain the stream APIs for container
From  - ReadStreamAsync,DeleteStreamAsync,ReplaceStreamAsync 
To  - ReadContainerStreamAsync,DeleteConatinerStreamAsync,ReplaceContainerStreamAsync 

Also it will fix the gap between actual APIs and example implementation in documentation on all the resources through out the code and add missing examples.

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 1]
```diff
@@ -66,14 +66,34 @@ public abstract class Container
...
+        /// A <see cref="Task"/> containing a <see cref="ResponseMessage"/> containing the read resource record.
+        /// <example>
+        /// <code language="c#">
+        /// <![CDATA[
+        /// Container container = this.database.GetContainer("containerId");
+        /// ResponseMessage response = await container.ReadContainerStreamAsync();
+        /// ]]>
+        /// </code>
+        /// </example>
+        /// </returns>
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs]: Returns should not include the example. It's normally summary, param, returns, examples

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 2]
```diff
@@ -112,6 +132,30 @@ public abstract class Container
...
+        ///
+        /// <code language="c#">
+        /// <![CDATA[
+        /// ContainerProperties containerProperties = containerReadResponse;
+        /// containerProperties.IndexingPolicy.Automatic = false;
+        /// ResponseMessage response = await container.ReplaceContainerStreamAsync(containerProperties);
+        /// ]]>
+        /// </code>
+        /// </example>
+        /// <returns>
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs]: Isn't return normally after params?

File: Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs [hunk 3]
```diff
@@ -194,6 +194,16 @@ public abstract class Database
         /// </summary>
         /// <param name="requestOptions">(Optional) The options for the container request <see cref="RequestOptions"/></param>
         /// <param name="cancellationToken">(Optional) <see cref="CancellationToken"/> representing request cancellation.</param>
+        /// <example>
+        /// <code language="c#">
+        /// <![CDATA[
+        /// //Reads a Database resource where
+        /// // - database_id is the ID property of the Database resource you wish to read.
+        /// Database database = this.cosmosClient.GetDatabase(database_id);
+        /// ResponseMessage response = await database.ReadContainerStreamAsync();
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs]: Database should not be able to read a container. Should this be database.ReadDatabaseStreamAsync?

File: Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs [hunk 4]
```diff
@@ -206,6 +216,16 @@ public abstract class Database
...
+        /// <![CDATA[
+        /// //Delete a Database resource where
+        /// // - database_id is the ID property of the Database resource you wish to delete.
+        /// Database database = this.cosmosClient.GetDatabase(database_id);
+        /// await database.DeleteContainerStreamAsync();
+        /// ]]>
+        /// </code>
+        /// </example>
         /// <returns>A <see cref="Task"/> containing a <see cref="ResponseMessage"/> which will contain information about the request issued.</returns>
         public abstract Task<ResponseMessage> DeleteStreamAsync(
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs]: Shouldn't this be updated to DeleteDatabaseStreamAsync?

File: Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs [hunk 5]
```diff
@@ -437,6 +459,20 @@ public abstract class Database
...
+        /// Creates a container as an asynchronous operation in the Azure Cosmos service and return stream response.
+        /// <code language="c#">
+        /// <![CDATA[
+        /// ContainerProperties containerProperties = new ContainerProperties()
+        /// {
+        ///     Id = Guid.NewGuid().ToString(),
+        ///     PartitionKeyPath = "/pk",
+        /// };
+        ///
+        /// ResponseMessage response = await this.cosmosDatabase.CreateContainerStreamAsync(containerProperties);
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs]: We should show that the response is IDisposable. We can either wrap it in a using statement or call dispose.

File: Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs [hunk 6]
```diff
@@ -455,6 +491,16 @@ public abstract class Database
         /// <param name="queryDefinition">The cosmos SQL query definition.</param>
         /// <param name="continuationToken">(Optional) The continuation token in the Azure Cosmos DB service.</param>
         /// <param name="requestOptions">(Optional) The options for the item query request <see cref="QueryRequestOptions"/></param>
+        /// <example>
+        /// This create the type feed iterator for containers with queryDefinition as input.
+        /// <code language="c#">
+        /// <![CDATA[
+        /// string queryText = "SELECT * FROM c where c.id like '%testId%'";
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs]: Let do an example with a parameter to show the benefit of query definition overload. ` "SELECT * FROM c where c.id like @testId"`

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 7]
```diff
@@ -786,9 +812,9 @@ public abstract class Container
         ///     public int cost {get; set;}
         /// }
         /// 
-        /// QueryDefinition sqlQuery = new QueryDefinition("select * from ToDos t where t.cost > @expensive").WithParameter("@expensive", 9000);
+        /// QueryDefinition queryDefinition = new QueryDefinition("select * from ToDos t where t.cost > @expensive").WithParameter("@expensive", 9000);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs]: .With.. in new line

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 8]
```diff
@@ -786,9 +812,9 @@ public abstract class Container
...
         /// FeedIterator<ToDoActivity> feedIterator = this.Container.GetItemQueryIterator<ToDoActivity>(
-        ///     sqlQueryDefinition: sqlQuery, 
+        ///     queryDefinition: queryDefinition,
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs]: namedParameters remove (okey to set null for CT).

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 9]
```diff
@@ -746,7 +772,7 @@ public abstract class Container
         /// 
         /// QueryDefinition queryDefinition = new QueryDefinition("select * from ToDos t where t.cost > @expensive").WithParameter("@expensive", 9000);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs]: .With... in new line

File: Microsoft.Azure.Cosmos/src/Resource/Scripts/Scripts.cs [hunk 10]
```diff
@@ -657,6 +737,16 @@ public abstract class Scripts
         /// <param name="queryDefinition">The cosmos SQL query definition.</param>
         /// <param name="continuationToken">(Optional) The continuation token in the Azure Cosmos DB service.</param>
         /// <param name="requestOptions">(Optional) The options for the item query request <see cref="QueryRequestOptions"/></param>
+        /// <example>
+        /// This create the type feed iterator for UDF with queryDefinition as input.
+        /// <code language="c#">
+        /// <![CDATA[
+        /// Scripts scripts = this.container.GetScripts();
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Scripts/Scripts.cs]: I thought scripts was a property now? `this.container.Scripts`

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 11]
```diff
@@ -746,7 +772,7 @@ public abstract class Container
...
         /// FeedIterator feedIterator = this.Container.GetItemQueryStreamIterator(
-        ///     queryDefinition: queryDefinition, 
+        ///     queryDefinition: queryDefinition,
         ///     requestOptions: new QueryRequestOptions { PartitionKey = new PartitionKey("Error")});
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs]: Remove named parameters

**Comment by kirankumarkolli** [comment]
>         ///     requestOptions: new QueryRequestOptions { PartitionKey = new PartitionKey("Error")});

named parameters

---
Refers to: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:858 in 516bf05. [](commit_id = 516bf057e1eea0a63521321dfa80c9b4279a1491, deletion_comment = False)

**Comment by kirankumarkolli** [comment]
>         ///     requestOptions: new QueryRequestOptions { PartitionKey = new PartitionKey("Error")});

named parameters

---
Refers to: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:899 in 516bf05. [](commit_id = 516bf057e1eea0a63521321dfa80c9b4279a1491, deletion_comment = False)

**Comment by kirankumarkolli** [comment]
>         public abstract ChangeFeedProcessorBuilder GetChangeFeedProcessorBuilder<T>(

example here would be great (next PR)

---
Refers to: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:1024 in 516bf05. [](commit_id = 516bf057e1eea0a63521321dfa80c9b4279a1491, deletion_comment = False)

**Comment by kirankumarkolli** [comment]
>         /// <returns>An instance of <see cref="ChangeFeedProcessorBuilder"/></returns>

Example in new PR

---
Refers to: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:1037 in 516bf05. [](commit_id = 516bf057e1eea0a63521321dfa80c9b4279a1491, deletion_comment = False)

File: Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs (hunk 3 above)
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs]: On Database there is no ambiguity on what read is for. On Container there is hence qualification. 
It should be database.ReadStreamAsync()

---
In reply to: [301773610](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/524#discussion_r301773610) [](ancestors = 301773610)

File: Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs [hunk 12]
```diff
@@ -206,6 +216,16 @@ public abstract class Database
         /// </summary>
         /// <param name="requestOptions">(Optional) The options for the container request <see cref="RequestOptions"/></param>
         /// <param name="cancellationToken">(Optional) <see cref="CancellationToken"/> representing request cancellation.</param>
+        /// <example>
+        /// <code language="c#">
+        /// <![CDATA[
+        /// //Delete a Database resource where
+        /// // - database_id is the ID property of the Database resource you wish to delete.
+        /// Database database = this.cosmosClient.GetDatabase(database_id);
+        /// await database.DeleteContainerStreamAsync();
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs]: DeleteStreamAsync

**Comment by kirankumarkolli** [comment]
>         /// you can read from it or write to it.

Please add text that Get doesn't guarantee existence.

---
Refers to: Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs:240 in 516bf05. [](commit_id = 516bf057e1eea0a63521321dfa80c9b4279a1491, deletion_comment = False)

File: Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs (hunk 12 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs]: Why should this be DeleteStreamAsync? Shouldn't it be DeleteDatabaseStreamAsync()?

**Comment by j82w** [comment]
@kirankumarkolli this isn't a named parameter. The bigger issue is it missing the "()" on the new object creation.

> > ```
> >     ///     requestOptions: new QueryRequestOptions { PartitionKey = new PartitionKey("Error")});
> > ```
> 
> named parameters
> 
> Refers to: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:899 in [516bf05](https://github.com/Azure/azure-cosmos-dotnet-v3/commit/516bf057e1eea0a63521321dfa80c9b4279a1491). [](commit_id = [516bf05](https://github.com/Azure/azure-cosmos-dotnet-v3/commit/516bf057e1eea0a63521321dfa80c9b4279a1491), deletion_comment = False)

File: Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs [hunk 13]
```diff
@@ -455,6 +491,16 @@ public abstract class Database
...
         /// <param name="continuationToken">(Optional) The continuation token in the Azure Cosmos DB service.</param>
         /// <param name="requestOptions">(Optional) The options for the item query request <see cref="QueryRequestOptions"/></param>
+        /// <example>
+        /// This create the type feed iterator for containers with queryDefinition as input.
+        /// <code language="c#">
+        /// <![CDATA[
+        /// string queryText = "SELECT * FROM c where c.id like '%testId%'";
+        /// QueryDefinition queryDefinition = new QueryDefinition(queryText);
+        /// FeedIterator<ContainerProperties> resultSet = this.cosmosDatabase.GetContainerQueryIterator<ContainerProperties>(queryDefinition);
+        /// ]]>
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs]: please include the iterator code also

File: Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs [hunk 14]
```diff
@@ -468,6 +514,16 @@ public abstract class Database
         /// <param name="queryDefinition">The cosmos SQL query definition.</param>
         /// <param name="continuationToken">The continuation token in the Azure Cosmos DB service.</param>
         /// <param name="requestOptions">(Optional) The options for the container request <see cref="QueryRequestOptions"/></param>
+        /// <example>
+        /// This create the stream feed iterator for containers with queryDefinition as input.
+        /// <code language="c#">
+        /// <![CDATA[
+        /// string queryText = "SELECT * FROM c where c.id like '%testId%'";
+        /// QueryDefinition queryDefinition = new QueryDefinition(queryText);
+        /// FeedIterator resultSet = this.cosmosDatabase.GetContainerQueryStreamIterator(queryDefinition);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs]: please include the iterator code also

File: Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs [hunk 15]
```diff
@@ -481,6 +537,15 @@ public abstract class Database
         /// <param name="queryText">The cosmos SQL query text.</param>
         /// <param name="continuationToken">(Optional) The continuation token in the Azure Cosmos DB service.</param>
         /// <param name="requestOptions">(Optional) The options for the item query request <see cref="QueryRequestOptions"/></param>
+        /// <example>
+        /// This create the type feed iterator for containers with queryText as input,
+        /// <code language="c#">
+        /// <![CDATA[
+        /// string queryText = "SELECT * FROM c where c.id like '%testId%'";
+        /// FeedIterator<ContainerProperties> resultSet = this.cosmosDatabase.GetContainerQueryIterator<ContainerProperties>(queryText);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs]: Another example for iterator with no arguments please

**Comment by kirankumarkolli** [comment]
>         /// Create an Azure Cosmos container through a Fluent API.

Creates a containerBuilder

---
Refers to: Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs:578 in 516bf05. [](commit_id = 516bf057e1eea0a63521321dfa80c9b4279a1491, deletion_comment = False)

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 16]
```diff
@@ -132,14 +176,32 @@ public abstract class Container
...
 
+        /// <summary>
+        /// Delete a <see cref="ContainerProperties"/> from the Azure Cosmos DB service as an asynchronous operation.
+        /// </summary>
+        /// <param name="requestOptions">(Optional) The options for the container request <see cref="RequestOptions"/></param>
+        /// <param name="cancellationToken">(Optional) <see cref="CancellationToken"/> representing request cancellation.</param>
+        /// <example>
+        /// <code language="c#">
+        /// <![CDATA[
+        /// Container container = this.database.Containers["containerId"];
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs]: We no longer have index over loads. It should be this.database.GetContainer("ContainerId");

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 17]
```diff
@@ -670,7 +696,7 @@ public abstract class Container
         /// Delete an item from Cosmos
         /// <code language="c#">
         /// <![CDATA[
-        /// using(CosmosResponseMessage response = this.container.DeleteItemStreamAsync(partitionKey: "itemPartitionKey", id: "itemId"))
+        /// using(ResponseMessage response = await this.container.DeleteItemStreamAsync("itemId", "itemPartitionKey"))
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs]: new PartitionKey("itemPartitionKey")

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 18]
```diff
@@ -920,19 +946,19 @@ public abstract class Container
         /// }
         ///  
         /// // Query by the Title property
-        /// Book book = container.Items.GetItemLinqQuery<Book>(allowSynchronousQueryExecution = true)
+        /// Book book = container.Items.GetItemLinqQuery<Book>(true)
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs]: Remove Items, it was merged with container

File: Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs [hunk 19]
```diff
@@ -215,9 +235,10 @@ public abstract class Database
         /// Returns a reference to a container object. 
         /// </summary>
         /// <param name="id">The cosmos container id.</param>
+        /// <returns>Cosmos container proxy</returns>
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Database/Database.cs]: nit: Use reference instead of proxy to be consistent with the below content.

**Pull request merged**